include/deal.II/lac/la_parallel_vector.h

```diff
@@ -175,13 +176,58 @@ namespace LinearAlgebra
      * fail in some circumstances. Therefore, it is strongly recommended to
      * not rely on this class to automatically detect the unsupported case.
      *
-     * @author Katharina Kormann, Martin Kronbichler, 2010, 2011, 2016
+     * <h4>CUDA support</h4>
+     *
+     * This vector class supports two different memory spaces: Host and CUDA. By
+     * default, the memory space is Host and all the data are allocated on the
+     * CPU. When the memory space is CUDA all the data is allocated on the GPU.
+     * The operations on the vector are performed on the chosen memory space. *
+     * From the host, there are two methods to access the elements of the Vector
+     * when using the CUDA memory space: <ul>
+     * <li> use get_values()
+     * <code>
+     * Vector<double, MemorySpace::CUDA> vector(local_range, comm);
+     * double* vector_dev = vector.get_values();
+     * std::vector<double> vector_host(local_range.n_elements(), 1.);
+     * Utilities::CUDA::copy_to_dev(vector_host, vector_dev);
+     * </code>
+     * <li> use import()
+     * <code>
+     * Vector<double, MemorySpace::CUDA> vector(local_range, comm);
+     * ReadWriteVector<double> rw_vector(local_range);
+     * for (auto & val : rw_vector)
+     *   val = 1.;
+     * vector.import(rw_vector, VectorOperations::insert);
+     * </code>
+     * </ul>
+     * The import method is a lot safer and will perform an MPI communication if
+     * necessary. Since an MPI communication may be performed, import needs to
+     * be called on all the processors.
+     *
+     * @note Vector using CUDA memory space are not instantiated by default so
```

Comment:
Any reasons for this? This is not our current policy of instantiating for all reasonable template arguments if we can. When we compile with `DEAL_II_WITH_CUDA=ON`, I would think it makes sense to also instantiate this class for `float` and `double` using `CUDA` memory space.

Reply:
The comment is wrong. At one point, I had problems to get the instantiation right but I've fixed it since and forgot to remove the comment.